server/src/com/cloud/network/vpn/Site2SiteVpnManagerImpl.java

```diff
@@ -426,72 +426,52 @@ public Site2SiteCustomerGateway updateCustomerGateway(UpdateVpnCustomerGatewayCm
         String ikePolicy = cmd.getIkePolicy();
-        String espPolicy = cmd.getEspPolicy();
         if (!NetUtils.isValidS2SVpnPolicy("ike", ikePolicy)) {
-            throw new InvalidParameterValueException("The customer gateway IKE policy" + ikePolicy + " is invalid!  Verify the required Diffie Hellman (DH) group is specified.");
+            throw new InvalidParameterValueException("Error while updating a customer gateway. The IKE policy" + ikePolicy + " is invalid! Verify whether the required Diffie Hellman (DH) group is specified.");
         }
+        String espPolicy = cmd.getEspPolicy();
         if (!NetUtils.isValidS2SVpnPolicy("esp", espPolicy)) {
             throw new InvalidParameterValueException("The customer gateway ESP policy" + espPolicy + " is invalid!");
         }
         Long ikeLifetime = cmd.getIkeLifetime();
-        if (ikeLifetime == null) {
-            // Default value of lifetime is 1 day
-            ikeLifetime = (long)86400;
-        }
-        if (ikeLifetime > 86400) {
+        if (ikeLifetime != null && ikeLifetime > 86400) {
             throw new InvalidParameterValueException("The IKE lifetime " + ikeLifetime + " of vpn connection is invalid!");
         }
         Long espLifetime = cmd.getEspLifetime();
-        if (espLifetime == null) {
-            // Default value of lifetime is 1 hour
-            espLifetime = (long)3600;
-        }
-        if (espLifetime > 86400) {
+        if (espLifetime != null && espLifetime > 86400) {
             throw new InvalidParameterValueException("The ESP lifetime " + espLifetime + " of vpn connection is invalid!");
         }
 
-        Boolean dpd = cmd.getDpd();
-        if (dpd == null) {
-            dpd = false;
-        }
-
-        Boolean encap = cmd.getEncap();
-        if (encap == null) {
-            encap = false;
-        }
-
         checkCustomerGatewayCidrList(guestCidrList);
 
-        long accountId = gw.getAccountId();
-        Site2SiteCustomerGatewayVO existedGw = _customerGatewayDao.findByNameAndAccountId(name, accountId);
-        if (existedGw != null && existedGw.getId() != gw.getId()) {
-            throw new InvalidParameterValueException("The customer gateway with name " + name + " already existed!");
+        String name = cmd.getName();
+        if (name != null) {
+            long accountId = gw.getAccountId();
+            Site2SiteCustomerGatewayVO existedGw = _customerGatewayDao.findByNameAndAccountId(name, accountId);
+            if (existedGw != null && existedGw.getId() != gw.getId()) {
+                throw new InvalidParameterValueException("The customer gateway with name " + name + " already existed!");
+            }
         }
 
         gw.setName(name);
         gw.setGatewayIp(gatewayIp);
         gw.setGuestCidrList(guestCidrList);
         gw.setIkePolicy(ikePolicy);
         gw.setEspPolicy(espPolicy);
-        gw.setIpsecPsk(ipsecPsk);
+        gw.setIpsecPsk(cmd.getIpsecPsk());
```

Comment:
@rhtyd so, I have to move these into the conditions otherwise they may be null right?